### DISCUSSION TITLE
FIX for Broken Search Available - Copilot4Eclipse 1.2.2, Please Read

### DISCUSSION THREAD

**Discussion by wayneparrott** [Genuitec/Copilot4Eclipse-community#23] [Help] [answered]
This note is a general notice for anyone experiencing broken Search facility in their Eclipse-based development environment.
User thread here: https://github.com/Genuitec/Copilot4Eclipse-community/discussions/19

**FIXED**
If the Search facility is broken in your Eclipse-based development environment (Cube) you will need to reinstall it followed by installing the latest Copilot4Eclipse v1.2.2 from the Eclipse Marketplace. We recommend that you install your new Eclipse-based dev environment to a new clean directory to avoid any old plugin conflicts.

We apologize for this inconvenience.

**DETAILS**
Thanks for sharing this info. We identified a really odd situation where somewhere (haven't isolated it) there is a loose specification on the eclipse search.core plugin. And because Copilot4Eclipse setup Eclipse to look for it's dependent plugins from the latest Eclipse distro it  resulted in the Eclipse provisioning system optimistically picking up a new search plugin version from the recent 2024-09 release. This created an incompatibility with the Eclipse search facility. 

**RECOVERY**
We updated the Copilot4Eclipse plugin constraints which seems to resolve the side-effect of the Eclipse search plugin from updating. We have released the Copilot4Eclipse v1.2.2 update to the Eclipse Marketplace earlier today. Unfortunately to recover the broken Search facility will require reinstalling your Eclipse-based dev environment.

**Comment by axbusch** [comment]
when i look in the marketplace i still only get 1.2.1 ?

![image](https://github.com/user-attachments/assets/db290698-261a-4ed9-be19-d7f2f3903c61)

**Reply by brianvfernandes** [reply, marked as answer]
Sorry about that - it's only the version "label" that was incorrect (now fixed), the update site it points to would give you 1.2.2 regardless.

**Comment by kxi-max** [comment]
Issue is not fixed
Updated to copilot v1.2.2 as suggested above, search still fails

  (Eclipse Version: 2024-06 (4.32.0) Build id: 20240606-1231)
  Copilot4Eclipse	1.2.2.202409201559

**Comment by axbusch** [comment]
Did you reinstall?

**Reply by kxi-max** [reply]
re-install Eclipse or copilot? I upgraded copilot plugin

**Reply by axbusch** [reply]
Reinstall eclipse und then copilot. Upgrade the plugin is not enough

**Reply by kxi-max** [reply]
Not using an installed version of eclipse, it's unzipped standalone eclipse.   What to do?  uninstall copilot and reinstall?

**Reply by toddewilliams** [reply]
1) Unzip Eclipse again into a completely new location. 

2) Run Eclipse from this new location.

3) Install Copilot4Eclipse

**Reply by kxi-max** [reply]
Tried that, the latest plugin version 1.3.0, which was released a couple of days ago, still has the same search error.   Any way to get to 1.2.2?   Only 1.3.0 shows in the list of options when installing

**Reply by wayneparrott** [reply]
@kxi-max I am helping look into your issue. Based on earlier info in this thread I created several clean installs of the Eclipse 2024-06 distros (java, jee, committers) with the latest Copilot4Eclipse 1.3.1 installed from the Marketplace. I have not observed any issues with Search (ctrl+H). 

I want to isolate this issue so I do the following:
1 - download and unzip Eclipse 4.32. This is the platform you stated you are using. I installed 3-4 different distros of this version.
2 - create a new workspace
3 - install copilot4eclipse from the Eclipse Marketplace (1.3.1 current)
4 - create a helloworld project
5 - Cti+h, search for "hello" and observe results in Search view

Did you try a clean installation of Eclipse 4.32 with only the latest earlier version of Copilot4Eclipse?

I would like to replicate your experience. 
1 - Please share the exact Eclipse version and distro you are using. If you are using an IDE built on Eclipse please share that info. 
2 - Share a list of all extra plugins you have may have installed from the Marketplace or elsewhere.
3 - Share any useful error log entries and a screenshot or two of error msg is always helpful
4 - Can you share a screenshot of this eclipse preference page: Window menu > Preferences > Install/Update > Available Software Sites

Re: #4, I'm curious if your Eclipse installation is picking up newer incompatible Search plugins from another site.

If you would rather not share some of this info here you can send it support@genuitec.com att Wayne and include a link to this thread.

Wayne
Copilot4Eclipse Team

**Reply by lrozenblyum** [reply]
Clean Eclipse install 2024-06 EE (4.32.0) + CoPilot plugin 1.3.1 - same search corruption reproduced (well there are some other plug-ins installed).

Please tell me if some more details needed.

NOTE: eclipse log contains some errors about CoPilot, yet unlikely they're related to the issue:

```
!ENTRY org.eclipse.jface 2 0 2024-10-30 17:23:09.304
!MESSAGE The image could not be loaded: URLImageDescriptor(platform:/plugin/com.genuitec.copilot4eclipse.chat/icons/message-outline-blue.png)
!STACK 0
org.eclipse.jface.resource.DeviceResourceException: Unable to create resource URLImageDescriptor(platform:/plugin/com.genuitec.copilot4eclipse.chat/icons/message-outline-blue.png)
	at org.eclipse.jface.resource.ImageDescriptor.createResource(ImageDescriptor.java:231)

```

**Reply by lrozenblyum** [reply]
@wayneparrott reported also an issue to m2e: https://github.com/eclipse-m2e/m2e-core/issues/1886
maybe it's related to some generic problem in Eclipse...

**Comment by kxi-max** [comment]
@wayneparrott   Thanks for looking into the issue.  

This a new installation,  No other plugins are installed after unzipping Eclipse except for copilot.  But I see that my copilot has 1.3.0. instead of the 1.3.1 you're using.   Let me try to upgrade to 1.3.1.  And will post here if the issue is resolved. 

Eclipse:   Version: 2024-06 (4.32.0), Build id: 20240606-1231
Copilot4Eclipse	1.3.0.202410241454	com.genuitec.copilot4eclipse.feature.feature.group	Genuitec, LLC

**Reply by kxi-max** [reply]
Confirming that search works with a clean installation of 2024-06 and Copilot4Eclipse 1.3.1.  Intersecting point: after Copilot 1.3.1 installation, installed TestNG 7.4.0.202106051955 plugin, then search breaks.  

However, if install TestNG first on a clean eclipse base, then install Copilort 1.3.1, search works.

**Reply by wayneparrott** [reply]
@kxi-max Thanks for sharing these details. I hope to replicate it and identify the root cause.
W

**Reply by lrozenblyum** [reply]
@wayneparrott you may find these findings interesting: https://github.com/eclipse-m2e/m2e-core/issues/1886#issuecomment-2465407302
(the same problem exists in m2e, looks to be caused by Eclipse bug: https://github.com/eclipse-platform/eclipse.platform.ui/pull/2510)

**Reply by wayneparrott** [reply]
@lrozenblyum thx for the heads-up. We have been following this as well. Good to see a fix at the  platform scope which resolves this issue for users of Copilot4Eclipse and other popular Eclipse plugins.
